torch/csrc/utils/python_arg_parser.h

```diff
@@ -179,7 +179,7 @@ struct PYBIND11_EXPORT PythonArgParser {
   bool traceable;
 };
 
-struct PYBIND11_EXPORT FunctionSignature {
+struct FunctionSignature {
```

Comment:
nit: can you add what you wrote about FunctionSignature being immutable as a comment here so we don't attempt to do the same thing in the future?